src/examples/ExamplesPanel.ts

```diff
@@ -82,41 +83,62 @@ export class ExamplesPlanel {
-                  const resultFolder = path.join(
-                    selectedFolder[0].fsPath,
-                    message.name
-                  );
-                  ensureDir(resultFolder)
-                    .then(() => {
-                      utils.copyFromSrcProject(
-                        message.project_path,
-                        resultFolder
-                      );
-                      const projectPath = vscode.Uri.file(resultFolder);
-                      vscode.commands.executeCommand(
-                        "vscode.openFolder",
-                        projectPath
-                      );
-                    })
-                    .catch((err) => {
-                      Logger.errorNotify("Error copying ESP-IDF example", err);
-                    });
-                } else {
-                  vscode.window.showInformationMessage("No folder selected");
-                }
+            const selectedFolder = await vscode.window.showOpenDialog({
+              canSelectFolders: true,
+              canSelectFiles: false,
+              canSelectMany: false,
+            });
+            if (!selectedFolder) {
+              return;
+            }
+            try {
+              const resultFolder = path.join(
+                selectedFolder[0].fsPath,
+                message.name
+              );
+              await ensureDir(resultFolder);
+              await utils.copyFromSrcProject(
+                message.project_path,
+                resultFolder
+              );
+              const settingsJsonPath = path.join(
+                resultFolder,
+                ".vscode",
+                "settings.json"
+              );
+              const settingsJson = await readJSON(settingsJsonPath);
+              const modifiedEnv = utils.appendIdfAndToolsToPath();
+              const idfTarget = modifiedEnv.IDF_TARGET || "esp32";
+              const compilerPath = await utils.isBinInPath(
+                `xtensa-${idfTarget}-elf-gdb`,
+                resultFolder,
+                modifiedEnv
+              );
+              settingsJson["C_Cpp.default.compilerPath"] = compilerPath;
+              await writeJSON(settingsJsonPath, settingsJson, {
+                spaces:
+                  vscode.workspace.getConfiguration().get("editor.tabSize") ||
+                  2,
               });
+              const projectPath = vscode.Uri.file(resultFolder);
+              vscode.commands.executeCommand("vscode.openFolder", projectPath);
+            } catch (error) {
+              const msg = `Error copying ESP-IDF example.`;
+              Logger.error(msg, error);
+              const opt = await vscode.window.showErrorMessage(
+                msg,
+                "Show Docs",
+                "Ok"
+              );
+              if (opt === "Show Docs") {
+                vscode.env.openExternal(vscode.Uri.parse(ESP.URL.Docs.README));
```

Comment:
LGTM.

But please create a issue for tracking that we replace the URL with proper one at later point of time